src/main/scala/com/github/unchama/buildassist/MenuInventoryData.scala

```diff
@@ -620,46 +390,64 @@ object MenuInventoryData {
     itemstack.setItemMeta(skullmeta)
     AsyncInventorySetter.setItemAsync(inventory, 45, itemstack)
 
+    // 4ページ目を開く
+    itemstack = new ItemStack(Material.SKULL_ITEM, 1, 3)
+    skullmeta = ItemMetaFactory.SKULL.getValue
+    skullmeta.setDisplayName(YELLOW.toString + "" + UNDERLINE + "" + BOLD + "4ページ目へ")
+    lore = List(RESET.toString + "" + DARK_RED + "" + UNDERLINE + "クリックで移動")
+    skullmeta.setLore(lore.asJava)
+    skullmeta.setOwner("MHF_ArrowDown")
+    itemstack.setItemMeta(skullmeta)
+    AsyncInventorySetter.setItemAsync(inventory, 53, itemstack)
+
     //石炭を消費して粘土をレンガに変換4～4000
-    {
-      val num_1 = playerdata_s.minestack.getStackedAmountOf(Util.findMineStackObjectByName("clay_ball"))
-      val num_2 = playerdata_s.minestack.getStackedAmountOf(Util.findMineStackObjectByName("brick_item"))
-      val num_3 = playerdata_s.minestack.getStackedAmountOf(Util.findMineStackObjectByName("coal"))
-      for (x <- 0 to 3) {
-        val p10 = power10(x)
-        itemstack = new ItemStack(Material.CLAY_BRICK, x)
-        itemmeta = Bukkit.getItemFactory.getItemMeta(Material.CLAY_BRICK)
-        itemmeta.setDisplayName(YELLOW.toString + "" + UNDERLINE + "" + BOLD + "石炭を消費して粘土をレンガに変換します")
-        lore = List(s"$RESET${GRAY}粘土${p10 * 4}個+石炭${p10}個→レンガ${p10 * 4}個", RESET.toString + "" + GRAY + "粘土の数:" + comma(num_1), RESET.toString + "" + GRAY + "石炭の数:" + comma(num_3), RESET.toString + "" + GRAY + "レンガの数:" + comma(num_2), RESET.toString + "" + GRAY + "建築LV" + BuildAssist.config.getMinestackBlockCraftlevel(3) + "以上で利用可能", RESET.toString + "" + DARK_RED + "" + UNDERLINE + "クリックで変換"
-        )
-        itemmeta.setLore(lore.asJava)
-        itemstack.setItemMeta(itemmeta)
-        inventory.setItem(x, itemstack)
-      }
-    }
+    batchCraftMenu(p,inventory,3,NonEmptyList.of(("clay_ball",4),("coal",1)),NonEmptyList.of(("brick_item",4)),0,3,0)
 
     //溶岩バケツを消費して粘土をレンガに変換50～5万
+    batchCraftMenu(p,inventory,3,NonEmptyList.of(("clay_ball",50),("lava_bucket",1)),NonEmptyList.of(("brick_item",50)),0,3,5)
+
+    inventory
+  }
+
+  private def batchCraftMenu(p: Player, inventory: Inventory, minestackBlockCraftlevel: Int, beforeItemID: NonEmptyList[(String, Int)], afterItemID: NonEmptyList[(String, Int)], minPower10Index: Int, maxPower10Index: Int, inventoryIndex: Int): Unit ={
+    //プレイヤーを取得
+    val player = p.getPlayer
+    //UUID取得
+    val uuid = player.getUniqueId
+    val playerdata_s = SeichiAssist.playermap(uuid)
+    var itemstack: ItemStack = null
+    var itemmeta: ItemMeta = null
+    var lore: List[String] = null
+    //順番に処理をする
+    val ingredients= beforeItemID
+      .toList
```

Comment:
この`toList`は改行しなくてもよいかなあ、と思います
```suggestion
    val ingredients= beforeItemID.toList
```